<a href="https://colab.research.google.com/github/DiyaRana7/Flyrank_ML/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## Lane: Refresh / Content Opportunity Scoring

I chose the **Refresh / Content Opportunity Scoring** lane because the goal is to help prioritize which content pages should be reviewed first. Large websites often contain thousands of pages, making manual review difficult and time-consuming. Instead of checking every page individually, I want to build a data-driven ranking system that uses observable search and engagement signals to identify pages that are most likely to benefit from review. This project supports human decision-making by helping teams focus their efforts on the highest-priority pages.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Research Question:**
Which content pages should be prioritized for refresh based on observable search and engagement signals?

### Decision
The decision is determining which content pages should be reviewed first for refresh, allowing the SEO team to prioritize limited time and resources effectively.

### Action
The SEO or content team can review the highest-ranked pages and decide whether to:
- update outdated content
- improve metadata
- expand information
- improve page structure
- monitor the page

### Cost of a Wrong Recommendation

**False Positive**
A page is recommended for refresh even though it does not need changes. This wastes time and resources.

**False Negative**
A page that actually needs refresh is missed. As a result, valuable content may continue losing traffic and visibility.

### Why Machine Learning?

Machine learning can analyze multiple signals such as impressions, CTR, content age, average position, and engagement together. Instead of manually reviewing thousands of pages, it helps rank pages based on their likelihood of needing review, making the decision process faster and more efficient.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os

if not os.path.exists("Flyrank_ML"):
    !git clone https://github.com/DiyaRana7/Flyrank_ML.git

%cd Flyrank_ML

Cloning into 'Flyrank_ML'...
remote: Enumerating objects: 96, done.
remote: Counting objects: 100% (96/96), done.
remote: Compressing objects: 100% (67/67), done.
remote: Total 96 (delta 18), reused 81 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (96/96), 1.84 MiB | 9.64 MiB/s, done.
Resolving deltas: 100% (18/18), done.
/content/Flyrank_ML/Flyrank_ML


In [ ]:
!ls data/raw

content_refresh_anonymized.csv


In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))

print("\nCorrelation between Search Volume and Impressions")
print(df["search_volume"].corr(df["impressions_90d"]))

print("\nMedian Word Count by Trend")
print(df.groupby("trend_direction")["word_count"].median())

print("\nTrend Direction Distribution")
print(df["trend_direction"].value_counts())

Total Pages: 30000

Correlation between Search Volume and Impressions
0.0012030310696208905

Median Word Count by Trend
trend_direction
down      2909.0
flat      2698.5
new       2239.0
stable    2912.5
up        2847.5
Name: word_count, dtype: float64

Trend Direction Distribution
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


### Observations

- The starter dataset contains **30,000 pages**, providing enough examples for exploratory analysis.
- The correlation between **search volume** and **impressions** is **0.0012**, showing that keyword search volume alone is not a strong predictor of page traffic.
- The median word count for **declining pages (2909 words)** and **growing pages (2847.5 words)** is very similar, suggesting that content length alone is unlikely to explain performance differences.
- The dataset contains **16,262 declining pages**, indicating that identifying refresh opportunities is a meaningful and practical problem for this project.

These observations suggest that multiple search and engagement signals should be considered together when deciding which pages deserve review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project does not attempt to predict Google's ranking algorithm or prove that refreshing a page will increase traffic.

Instead, it uses observable historical search and engagement signals to identify pages that may deserve human review.

The recommendations produced by this project are intended to support decision-making rather than replace human judgment.

Any relationships identified in this analysis are observational and should not be interpreted as proof of causation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## Self-Check

- ✅ I selected a project lane.
- ✅ I defined a clear research question.
- ✅ I explained the decision my project supports.
- ✅ I described the action someone can take.
- ✅ I explained the cost of incorrect recommendations.
- ✅ I used real numbers from the starter dataset.
- ✅ I used careful language and avoided causal claims.
- ✅ I understand that this project supports human decisions rather than replacing them.